docs/DeveloperGuide.md

```diff
@@ -1,34 +1,670 @@
+ the parameters required to execute. Once this condition is fulfilled, it returns a `Command` object to **LifEasier**
+  for the commands to be executed.
+  
+  
+#### 3.2.4 Command Component
+
+The `Command` component consists of an abstract `Command` class and the many different commands that inherit the 
+`Command` class. What is important to note is that each type of command class (e.g. `addLessonCommand`, `showNotesCommand`)
+ implements an abstract `execute()` method that carries out the command. Figure 3.2.4-1 below shows the classes that 
+ inherit from the `Command` class.
+ 
+ ![Figure 3.2.4-1](images/DeveloperGuide/Figure%203.2.4-1.png)  
+ _Figure 3.2.4-1: The classes that inherit from the Command class_
+ 
+#### 3.2.5 Model Component
+
+The `Model` component contains all the user's notes and tasks in memory. It is comprised of the `TaskList`, `NoteList`, `TaskHistory` and `NoteHistory` classes.
+The `TaskList` class handles additions, edits and deletions to the user's tasks, while the `TaskHistory` class allows users to undo their most recent task edits.
+Similarly, the `NoteList` class handles additions, edits and deletions to the user's notes, while the `NoteHistory` class allows users to undo their most recent notes edits.
+All classes in the `Model` component interact heavilt with the `Command` component as most commands make use of the user's tasks and notes.
+ 
+#### 3.2.6 Storage Component
+
+The `Storage` component handles saving of the users’ notes and tasks to persistent storage.
+ It does this after every addition, change, or deletion to the `TaskList` component or `NoteList` component. 
+ It also handles the moving of the stored `notes` to a separate `archive` file if instructed.
+
+## 4.0 Implementation
+
+The following section will go into further details on how the different features of **LifEasier** were implemented.
+
+### 4.1 Adding Lessons, Events, Deadlines (Fairuz)
+
+The `addLesson`, `addEvent` and `addDeadline` commands adds the specific task into the `TaskList`.
+
+##### Implementation
+
+Due to the different parameters required by the addition of different types of tasks, each command has
+ a specific intended parameter to be input by the user.
+
+For example, if the user uses the `addEvent` command, the user will have to input the `EVENT_NAME`,
+ `START_TIME` and `END_TIME`, whereas if the user uses the `addDeadline` command, the user will have input `DEADLINE_NAME` and `BY`.
+
+When the command is called, the class invokes the `TaskList` class method to add the specific task to
+ the `TaskList`, after which `Storage` is called to save the updated `TaskList` to the local save file. Figure 4.1-1 shows the 
+ sequence diagram for the `addDeadlineCommand`. This sequence diagram is applicable to the other two `addEventCommand` and 
+ `addLessonCommand`, with the only difference being in their parameters.
+ 
+ ![Figure 4.1-1](images/DeveloperGuide/Figure%204.1-1.png)  
```

Comment:
This is a really well drawn diagram! Quite easy to understand. I was just wondering, should addDeadline() have 3 parameters instead of 2?
![image](https://user-images.githubusercontent.com/60306071/97667896-5481a380-1a9a-11eb-9bd4-a649dd6de0cf.png)